### ISSUE TITLE
Getting reports for what is already done?

### ISSUE THREAD

**Issue by AbhineetSharmax** [serenity-js/serenity-js#355]
Hi @jan-molak ,

Greetings..

We have 450 scenarios which runs in parallel in Jenkins and and it takes around 3 hours for the completion and at last As post build action running npm run report for the report generation. 

Is it possible to get the reports and screenshots in between the test runs? Please suggest.

**Comment by abhinaba1080** [comment]
Hi, serenity-js creates granular JSON/xml reports for each scenario as well as takes the screenshot for them. You can get that granular JSON reports and screenshot  in completion of any scenario. But, the files names are system generated , so fetching a file and screenshot related to a specific scenario will be little tricky. 

If you need the HTML dashboard in between the tests, It is not possible as Serenity-CLI consolidates  those granular JSON reports after the whole test execution. 

One workaround may be that you can run   **serenity run** command after each scenario to create the dashboard and wait other scenarios untill this dashboard creation activity ends. But it will greatly increase your execution time. 
Or you can create a mechanism to fetch the granular json reports + screenshots separately and continuously make the dashboard in a separate parallel Jenkins job. 

Let me know if you find a hack for the same.

**Comment by jan-molak** [comment]
Hi @AbhineetSharmax !

[Serenity/JS version 2.0](https://serenity-js.org) has been released yesterday and introduced a new mechanism for generating and storing the reports, which results in much easier to identify (yet unique) file names

![image](https://user-images.githubusercontent.com/1089173/73736510-c958c480-4738-11ea-9ea3-f2e64d5d4744.png)

This, together with what @abhinaba1080 suggested above should help you address your scenario.

Please let us know if it did :)
